qiskit/pulse/builder.py

```diff
@@ -451,7 +452,8 @@ def call_schedule(self, schedule: Schedule):
     def call_subroutine(self,
                         subroutine: Union[circuit.QuantumCircuit, Schedule],
                         name: Optional[str] = None,
-                        **kw_params):
+                        value_dict: Optional[Dict[ParameterExpression, ParameterValueType]] = None,
+                        **kw_params: ParameterValueType):
```

Comment:
I wonder if these should be swapped? Should we force the user to specify `call(routine, value_dict={...})`, I lean towards yes as in the future if there was a canonical ordering for args we could do `call(routine, arg0, ..., argn=val)` with both args and kwargs. Do you have thoughts on this?

Reply:
You mean adding `*args: ParameterValeType` ? Could you please elaborate on _swapped_? I think we cannot define keyword augment (`kw_params`) before the positional argument (`value_dict`) and `ParameterExpression` object cannot be a keyword. I think adding `*args` is good idea. However `Schedule.parameters` is a `Set` object and I don't think the order is preserved.

Reply:
Resolving in favour of adding the notion of a function in a future PR.